packages/catlog/src/one/category.rs

```diff
@@ -178,114 +178,94 @@ object generators (unless there are nontrivial equations between objects), but
 can have infinitely many morphisms.
  */
 pub trait FgCategory: Category {
-    /// Is the object a generator of the category? Implies `self.has_ob(x)`.
-    fn has_ob_generator(&self, x: &Self::Ob) -> bool;
+    /// The type of object generators.
+    type ObGen: Eq + Into<Self::Ob>;
 
-    /// Is the morphism a generator of the category? Implies `self.has_hom(f)`.
-    fn has_hom_generator(&self, f: &Self::Hom) -> bool;
+    /// The type of morphism generators. Often Mor = Path<Ob, MorGen>
+    type MorGen: Eq + Into<Self::Mor>;
 
-    /// Iterates over object generators of the category.
-    fn ob_generators(&self) -> impl Iterator<Item = Self::Ob>;
-
-    /// Iterates over morphism generators of the category.
-    fn hom_generators(&self) -> impl Iterator<Item = Self::Hom>;
-
-    /// Iterates over morphism generators with the given domain.
-    fn generators_with_dom(&self, x: &Self::Ob) -> impl Iterator<Item = Self::Hom>;
-
-    /// Iterates over morphism generators with the given codomain.
-    fn generators_with_cod(&self, x: &Self::Ob) -> impl Iterator<Item = Self::Hom>;
+    /// The graph of generating morphisms and objects
+    fn generating_graph(&self) -> &impl FinGraph<V = Self::ObGen, E = Self::MorGen>;
```

Comment:
With our revised understand of the concept, an `FgCategory` doesn't have a generating graph. It has something in the spirit of a computad: a set of object generators and an endo-span of morphism generators on *objects* (not object generators).

So I think that `FgCategory` and `DblModel` need an interface for generators that does not make direct reference to graphs. OTOH, *discrete* double models do have a generating graph and the struct `DiscreteDblModel` should have a `generating_graph` method.

Reply:
Yes, OK, good point! Will update with regard to this.